semgrep/semgrep/test.py

```diff
@@ -187,23 +189,34 @@ def score_output_json(
                 continue
 
     for result in json_out["results"]:
-        reported_lines[str(Path(result["path"]).resolve())][result["check_id"]].append(
-            int(result["start"]["line"])
-        )
+        path = str(Path(result["path"]).resolve())
+        check_id = result["check_id"]
+        start_line = int(result["start"]["line"])
+        reported_lines[path][check_id].append(start_line)
+
+    test_lines: Dict[str, Set] = collections.defaultdict(set)
+    for lines in [ruleid_lines, ok_lines]:
+        for file_path, test_annotations in lines.items():
+            test_lines[file_path].update(test_annotations.keys())
+
+    rule_id_mismatch = False
+    if reported_lines:
+        for file_path, test_ids in test_lines.items():
+            reported_ids = set(reported_lines[file_path].keys())
+            if test_ids.symmetric_difference(reported_ids):
```

Comment:
I guess `reported_lines` it's the errors that Semgrep reported, and `test_lines` includes both `ruleid:` as well as `ok:` lines. If we have one rule for which there is no `ruleid:` but there is some `ok:`, then that rule will not be in `reported_lines` but it will be in `test_lines`, then we raise an error. Is my reasoning correct?

Reply:
Hmm, that is an interesting edge case. I believe you are correct. IMO it's still okay to error out because it'd be quite odd to have a test case with `ok`'s but no `ruleid`'s. We're also maintaining the existing behavior, so we can consider this error case more closely if a user has concerns :+1: